src/main/java/lotto/domain/Result.java

```diff
@@ -0,0 +1,45 @@
+package lotto.domain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class Result {
+    private final Map<Integer, Integer> result;
+
+    public Result(List<Lotto> lottos, Lotto winningLotto) {
+        result = new HashMap<>();
+        for (Lotto lotto : lottos) {
+            int count = lotto.countSameNo(winningLotto);
+            increaseNumOfLotto(count);
+        }
+    }
+
+    public Money getProfit() {
+        int profit = 0;
+        for (Jackpot jackpot : Jackpot.values()) {
+            profit += jackpot.getPrizeMoney() * getNumOfLotto(jackpot);
+        }
+        return new Money(profit);
+    }
+
+    public int getNumOfLotto(Jackpot jackpot) {
+        return getNumOfLotto(
+                jackpot.getNumOfCorrected()
+        );
+    }
+
+    private int getNumOfLotto(int numOfCorrected) {
+        return Optional.ofNullable(
+                result.get(numOfCorrected)
+        ).orElseGet(() -> 0);
+    }
+
+    void increaseNumOfLotto(int numOfCorrected) {
+        result.put(
+                numOfCorrected,
+                getNumOfLotto(numOfCorrected) + 1
+        );
+    }
```

Comment:
두 메서드를 하나로 줄여볼 수 있을 것 같아요 :)
```suggestion
    private void increaseNumOfLotto(int numOfCorrected) {
        result.put(numOfCorrected, result.getOrDefault(numOfCorrected, 0) + 1);
    }
```